crates/exec-wasmtime/src/lib.rs

```diff
@@ -51,10 +68,36 @@ pub fn execute() -> anyhow::Result<()> {
     // The FD is managed by the host or its parent.
     forget(host);
 
-    let args = toml::from_str::<Args>(&args).context("failed to decode arguments")?;
-
-    execute_with_args(args)?;
-
+    let Args {
+        log_level,
+        #[cfg(feature = "bench")]
+        profile,
+        package,
+    } = toml::from_str(&args).context("failed to decode arguments")?;
+
+    #[cfg(feature = "bench")]
+    let (flame_layer, _guard) = if let Some(profile) = profile {
+        use std::fs::File;
+        let profile = unsafe { File::from_raw_fd(profile) };
+        let flame_layer = tracing_flame::FlameLayer::new(profile);
+        let guard = flame_layer.flush_on_drop();
+        (Some(flame_layer), Some(guard))
+    } else {
+        (None, None)
+    };
+    let log_level: LevelFilter = log_level.map(Into::into).into();
+    {
+        let fmt_layer = tracing_subscriber::fmt::layer()
+            // TODO: Default to a secure log target
+            // https://github.com/enarx/enarx/issues/1042
+            .with_writer(std::io::stderr)
+            .with_filter(log_level);
+        let registry = registry().with(fmt_layer);
+        #[cfg(feature = "bench")]
+        let registry = registry.with(flame_layer);
+        let _guard = registry.set_default();
```

Comment:
What are these two guards guarding, and why are we having one shadow the other?

Reply:
The outermost needs to be dropped to ensure the profile is written to the file. Innermost reverts the log setting once it's dropped, which is relevant for nil backends where the log may have been previously initialized